ci/do_ci.sh

```diff
@@ -222,10 +222,10 @@ elif [[ "$CI_TARGET" == "bazel.gcc" ]]; then
   setup_gcc_toolchain
 
   echo "Testing ${TEST_TARGETS[*]}"
-  bazel_with_collection test "${BAZEL_BUILD_OPTIONS[@]}" -c fastbuild "${TEST_TARGETS[@]}"
+  bazel_with_collection test "${BAZEL_BUILD_OPTIONS[@]}" -c dbg "${TEST_TARGETS[@]}"
 
   echo "bazel release build with gcc..."
-  bazel_binary_build fastbuild
+  bazel_binary_build release
```

Comment:
We should do dbg as well. Otherwise it is full rebuild again.

Reply:
Sure, done.

Reply:
I'd prefer if we used `-c dbg`, since there are no debug builds done on the CI (...well, there is an ASan build with `-c dbg`, but a lot of projects turn off certain features in builds with sanitizers to avoid false-positives), and there were a few issues in the past that would have been caught by it.

Reply:
Makes sense.  Rolled back the switch to release, I'll wait for the review of https://github.com/envoyproxy/envoy/pull/16132 to complete.